xacro.cmake

```diff
@@ -87,5 +87,5 @@ MACRO(GENERATE_URDF_FILE FILENAME EXTENSION)
     SET(_XACRO_FILE_ ${CMAKE_CURRENT_SOURCE_DIR}/${FILENAME}.xacro)
   ENDIF (NOT EXISTS ${CMAKE_CURRENT_SOURCE_DIR}/${FILENAME}.xacro)
 
-  RUN_XACRO(${_XACRO_FILE_} ${CMAKE_CURRENT_BINARY_DIR}/${OUTPUT_FILE})
+  RUN_XACRO(${_XACRO_FILE_} ${CMAKE_CURRENT_BINARY_DIR}/${FILENAME}.${EXTENSION})
```

Comment:
How the hell did this ever worked !

Reply:
This is an excellent question. I guess a commit forgot to update this too at some point.

Reply:
or maybe xacro is able to understand that when the output is a directory, it just has to put a file with `${FILENAMEE}.urdf` in this directory.